lib/dai-plugin-mcd/src/SavingsService.js

```diff
@@ -0,0 +1,80 @@
+import { PublicService } from '@makerdao/services-core';
+import { ServiceRoles } from './constants';
+import { MDAI } from './index';
+import BigNumber from 'bignumber.js'
+import { RAY, WAD } from './constants'
+
+export default class SavingsService extends PublicService {
+  constructor(name = ServiceRoles.SAVINGS) {
+    super(name, ['smartContract', 'proxy', 'accounts', ServiceRoles.SYSTEM_DATA]);
+  }
+
+  async join(amountInDai) {
+    await this.get('proxy').ensureProxy();
+
+    const chi = await this.chi();
+    const amount = amountInDai.toBigNumber().div(chi);
+    return await this._proxyActions.dsrJoin(
+      this._daiAdapterAddress,
+      this._potAddress,
+      MDAI(amount).toFixed('wei'),
+      { dsProxy: true }
+    );
+  }
+
+  async exit(amountInDai) {
+    await this.get('proxy').ensureProxy();
+
+    const chi = await this.chi();
+    const amount = amountInDai.toBigNumber().div(chi)
+    return await this._proxyActions.dsrExit(
+      this._daiAdapterAddress,
+      this._potAddress,
+      MDAI(amount).toFixed('wei'),
+      { dsProxy: true }
+    );
+  }
+
+  async balance() {
+    const proxy = await this.get('proxy').ensureProxy();
+
+    return await this.balanceOf(proxy)
+  }
+
+  async balanceOf(guy) {
+    const amount = new BigNumber(await this._pot.pie(guy)).div(WAD)
+    const chi = await this.chi()
+    return MDAI(amount.times(chi))
+  }
+
+  async getTotalDai() {
+    const totalPie = new BigNumber(await this._pot.Pie()).div(WAD)
+    const chi = await this.chi()
+    return MDAI(totalPie.times(chi))
+  }
+
+  async getDaiSavingsRate() {
+    const dsr = await this._pot.dsr()
```

Comment:
I think `dsr` in this case is "interest rate per second." My sense is that it'd be more conventional to return an annual rate -- do you agree? (we could transform it by doing `dsr ^ secondsPerYear` like we do [with the gov fee](https://github.com/makerdao/dai.js/blob/37e38deeaf9414d26fd7d0c8ba5cf77926b3d33a/src/eth/EthereumCdpService.js#L375).)

Reply:
That's definitely true. I think we can rename `getDaiSavingsRate` to something like `getYearlyInterest` as well.